src/tasks.py

```
@@ -11,32 +12,47 @@
+        model: Model name. If None, uses runtime_config.
     """
 
     brief = CampaignBrief(**brief_dict)
 
+    # Use passed provider/model or fall back to runtime config
+    selected_provider = provider or runtime_config.provider
+    selected_model = model or runtime_config.model
+
     async def _run() -> Dict[str, Any]:
         if not settings.OPENAI_API_KEY:
             raise RuntimeError("OPENAI_API_KEY not configured")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)